registry/config.go

```diff
@@ -63,6 +65,10 @@ var (
 	emptyServiceConfig = newServiceConfig(ServiceOptions{})
 )
 
+var (
+	validHostnameRegex = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`)
```

Comment:
That said, we don't currently allow `_` in domains in the reference grammar, so there's no scenario where a registry with an `_` in the name would actually be usable.

I wonder if we should export `DomainRegexp` in the reference package and use that instead of defining a new regexp here (cc @dmcgowan).